### PR TITLE
8241504: Expose MemoryLayout annotations/attributes in the public API (foreign-jextract part)

### PR DESCRIPTION
**Pull request by JornVernee**
Hi,

These are the jextract changes needed after exposing layout attributes in the public API. This also fixes the pending merge conflict, which was closely related to the needed changes.

Jorn

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [JDK-8241504](https://bugs.openjdk.java.net/browse/JDK-8241504): Expose MemoryLayout annotations/attributes in the public API

### Reviewers
 * Maurizio Cimadamore ([mcimadamore](@mcimadamore) - Committer)

### Download
`$ git fetch https://git.openjdk.java.net/panama-foreign pull/69/head:pull/69`
`$ git checkout pull/69`

### CODE REVIEW

**Comment by JornVernee** [comment]
So, this includes a merge for: https://github.com/openjdk/panama-foreign/pull/68 and some other jextract fixes. Both are needed to get foreign-jextract building & passing tests again. Just the other changes can be seen when looking at the diff for the last commit: https://github.com/openjdk/panama-foreign/pull/69/commits/a1074729fb3190614f2e59947dfbed492e968ad3

File: test/jdk/java/foreign/NativeTestHelper.java [hunk 1]
```diff
@@ -38,19 +38,15 @@
     public static final SystemABI ABI = Foreign.getInstance().getSystemABI();
 
     public static boolean isIntegral(MemoryLayout layout) {
-        var optAbiType = layout.abiType();
-        if (!optAbiType.isPresent()) {
-            return false;
-        }
-        return switch(optAbiType.get()) {
+        return switch(SystemABI.Type.fromLayout(layout)) {
```
Comment (slowhog) [inline test/jdk/java/foreign/NativeTestHelper.java]: Need to catch the exception?

Comment (JornVernee) [reply test/jdk/java/foreign/NativeTestHelper.java]: No, I don't think so. The intent is to have a hard failure if the ABI type is missing. The old code was failing more silently by returning `false`.

Comment (JornVernee) [reply test/jdk/java/foreign/NativeTestHelper.java]: Similar motivation here: https://github.com/openjdk/panama-foreign/pull/64#discussion_r398488164

Comment (slowhog) [reply test/jdk/java/foreign/NativeTestHelper.java]: OK, I didn't look at use cases for this method, just thinking of compatibility of existing code. If the cases are expecting native ABI type to be available all the time, then yes, I agree.

**Pull request closed without merging**